matrix/buffer.py

```diff
@@ -1004,10 +1049,10 @@ def add_user(self, user_id, date, is_state, force_add=False):
         elif user.user_id.startswith("@freenode_"):
             short_name = shorten_sender(user.user_id[9:])
 
-        # TODO make this configurable
-        if not short_name or short_name in self.displayed_nicks.values():
-            # Use the full user id, but don't include the @
-            nick = user_id[1:]
```

Comment:
Could not quite wrap my head around this `if` branch.
Was this an workaround to remote ambiguity? I did not notice that this branch is ever executed.